src/main/java/org/humancellatlas/ingest/project/ProjectService.java

```diff
@@ -175,4 +182,33 @@ private ProjectBag gather(Project project) {
         return new ProjectBag(projects, envelopes);
     }
 
+    public Page<Project> filterProjects(String search, String wranglingState, String wrangler, Pageable pageable) {
+        List<Criteria> criterias = new ArrayList<>();
+        criterias.add(Criteria.where("isUpdate").is(false));
+        Query query = new Query();
+
+        if (wranglingState != null) {
+            criterias.add(Criteria.where("wranglingState").is(wranglingState));
+        }
+
+        if (wrangler != null) {
+            criterias.add(Criteria.where("primaryWrangler").is(wrangler));
+        }
+
+        if (search != null) {
+            criterias.add(new Criteria().orOperator(
+                    Criteria.where("content.project_core.project_title").regex(search, "i"),
+                    Criteria.where("content.project_core.project_description").regex(search, "i"),
+                    Criteria.where("content.project_core.project_short_name").regex(search, "i")
+            ));
+        }
+
+        query.addCriteria(new Criteria().andOperator(criterias.toArray(new Criteria[criterias.size()])));
+
+        System.out.println(query.toString());
```

Comment:
Are we logging to `System.out`?

Reply:
I agree with @amnonkhen I would instantiate a logger like this:
`private static final Logger log = LoggerFactory.getLogger(ProjectService.class);`
and would use it here like this:
`log.debug("Search query: " + query.toString())`